File: backend/src/main/java/com/example/backend/service/ShareService.java
package com.example.backend.service;

import com.example.backend.model.request.ShareRequest;
import com.example.backend.model.respon.ShareResponse;

public interface ShareService {
    // chia sẻ bài viết
    ShareResponse sharePost(Long userId, ShareRequest request);

    // xóa bài chia sẻ
    Boolean deleteShare(Long userId, Long shareId);
}

File: frontend/src/service/impl/ShareService.ts
import { http } from "../http/http";
import type { ShareRequest } from "../../utils/RequestType";
import type { ApiDataResponse, ShareResponse } from "../../utils/ResponType";

export const sharePost = (currentUserId: number, request: ShareRequest) => {
    return http.post<ApiDataResponse<ShareResponse>>(`/shares?currentUserId=${currentUserId}`, request);
};

export const deleteShare = (currentUserId: number, shareId: number) => {
    return http.delete<ApiDataResponse<boolean>>(`/shares/${shareId}?currentUserId=${currentUserId}`);
};
